src/acpi.rs

```diff
@@ -419,3 +424,357 @@ pub fn process_msct() -> (
+            ((*entry).DeviceHandle >> 8) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 12) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 16) & ACPI_NFIT_DEVICE_HANDLE_12BIT_MASK,
+            (*entry).PhysicalId,
+            (*entry).RegionId,
+            (*entry).RangeIndex,
+            (*entry).RegionIndex,
+            (*entry).RegionSize,
+            (*entry).RegionOffset,
+            (*entry).Address,
+            (*entry).InterleaveIndex,
+            (*entry).InterleaveWays,
+            (*entry).Flags,
+        );
+    }
+}
+
+// NFIT subtable type = 0x2
+fn parse_nfit_interleave_structure(entry: *const ACPI_NFIT_INTERLEAVE) {
+    unsafe {
+        assert_eq!((*entry).Header.Type, 2);
+        debug!(
+            "ACPI_NFIT_TYPE_INTERLEAVE - Type {}, Length {}
+            InterleaveStructureIndex: {:#x}
+            NumberOfLinesDescribed: {:#x}
+            LineOffset: {:#x}
+            LineSize: {:#x}",
+            (*entry).Header.Type,
+            (*entry).Header.Length,
+            (*entry).InterleaveIndex,
+            (*entry).LineCount,
+            (*entry).LineOffset[0],
+            (*entry).LineSize
+        );
+    }
+}
+
+// NFIT subtable type = 0x4
+fn parse_nfit_control_region_structure(entry: *const ACPI_NFIT_CONTROL_REGION) {
+    unsafe {
+        assert_eq!((*entry).Header.Type, 4);
+        debug!(
+            "ACPI_NFIT_TYPE_CONTROL_REGION - Type {}, Length {}
+                RegionIndex: {},
+                VendorId: {},
+                DeviceId: {},
+                RevisionId: {},
+                SubsystemVendorId: {},
+                SubsystemDeviceId: {},
+                SubsystemRevisionId: {},
+                Reserved: {:?},
+                SerialNumber: {},
+                Code: {},
+                Windows: {},
+                WindowSize: {},
+                CommandOffset: {},
+                CommandSize: {},
+                StatusOffset: {},
+                StatusSize: {},
+                Flags: {}",
+            (*entry).Header.Type,
+            (*entry).Header.Length,
+            (*entry).RegionIndex,
+            (*entry).VendorId,
+            (*entry).DeviceId,
+            (*entry).RevisionId,
+            (*entry).SubsystemVendorId,
+            (*entry).SubsystemDeviceId,
+            (*entry).SubsystemRevisionId,
+            (*entry).Reserved,
+            (*entry).SerialNumber,
+            (*entry).Code,
+            (*entry).Windows,
+            (*entry).WindowSize,
+            (*entry).CommandOffset,
+            (*entry).CommandSize,
+            (*entry).StatusOffset,
+            (*entry).StatusSize,
+            (*entry).Flags,
+        );
+    }
+}
+
+// NFIT subtable type 0x6
+fn parse_nfit_flush_hint_structure(entry: *const ACPI_NFIT_FLUSH_ADDRESS) {
+    unsafe {
+        assert_eq!((*entry).Header.Type, 6);
+        debug!(
```

Comment:
Can we just implement debug on `ACPI_NFIT_FLUSH_ADDRESS` and types from above to put actual print logic there?

Reply:
All these ACPI_NFIT_* types come from acpica-sys crate, so, we need to change it there. 

I am keeping the debug statements, for now, so that we can simply move the information to structures without going through ACPI docs again. Please let me know if you want to comment out these lines.